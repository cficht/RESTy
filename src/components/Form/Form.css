.Form {
  width: 20%;
  background-color: darkblue;
  color: white;
  padding: 20px;
  margin: 20px;

  textarea {
    display: block;
    margin: auto;
    width: 100%;
  }
  
}
