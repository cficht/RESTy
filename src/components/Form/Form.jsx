import React from 'react';
// import PropTypes from 'prop-types';
import styles from './Form.css';

const Form = () => (
  <div className={styles.Form}>
    <form>
      <label>Url:<input type="text"></input></label> 
      <div>
        <label><input type="radio" name="method" value="get" />GET</label>
        <label><input type="radio" name="method" value="post" />POST</label>
        <label><input type="radio" name="method" value="put" />PUT</label>
        <label><input type="radio" name="method" value="patch" />PATCH</label>
        <label><input type="radio" name="method" value="delete" />DELETE</label>
      </div>
      <textarea placeholder="Body"></textarea>
      <button type="submit">Submit</button>
    </form>
  </div>
);

Form.propTypes = {

};

export default Form;
