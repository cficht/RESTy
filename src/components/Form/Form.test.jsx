import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form component', () => {
  it('renders Form', () => {
    const wrapper = shallow(<Form 
      url=""
      body=""
      onChange={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
