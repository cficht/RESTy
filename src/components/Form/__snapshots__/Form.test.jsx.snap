// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component renders Form 1`] = `
<div
  className="Form"
>
  <form>
    <label>
      URL:
      <input
        name="url"
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <div>
      <label>
        <input
          name="method"
          onChange={[Function]}
          type="radio"
          value="GET"
        />
        GET
      </label>
      <label>
        <input
          name="method"
          onChange={[Function]}
          type="radio"
          value="POST"
        />
        POST
      </label>
      <label>
        <input
          name="method"
          onChange={[Function]}
          type="radio"
          value="PUT"
        />
        PUT
      </label>
      <label>
        <input
          name="method"
          onChange={[Function]}
          type="radio"
          value="PATCH"
        />
        PATCH
      </label>
      <label>
        <input
          name="method"
          onChange={[Function]}
          type="radio"
          value="DELETE"
        />
        DELETE
      </label>
    </div>
    <textarea
      name="body"
      onChange={[Function]}
      placeholder="Body"
      value=""
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
